packages/angular_devkit/build_webpack/test/utils/test-project-host.ts

```diff
@@ -0,0 +1,141 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  Path,
+  PathFragment,
+  getSystemPath,
+  normalize,
+  resolve,
+  virtualFs,
+} from '@angular-devkit/core';
+import { NodeJsSyncHost } from '@angular-devkit/core/node';
+import { SpawnOptions, spawn } from 'child_process';
+import { Stats } from 'fs';
+import { Observable } from 'rxjs/Observable';
+import { empty } from 'rxjs/observable/empty';
+import { concatMap, map } from 'rxjs/operators';
+
+
+interface ProcessOutput {
+  stdout: string;
+  stderr: string;
+}
+
+export class TestProjectHost extends NodeJsSyncHost {
+  private _syncHost: virtualFs.SyncDelegateHost<Stats>;
+
+  constructor(protected _root: Path) {
```

Comment:
This can inherit `ScopedHost` instead and `super(new NodeJsSyncHost, root)`, greatly simplifying this whole class.

Reply:
Discussed offline.